server/routes/damages/detailsOfDamages.ts

```diff
@@ -24,29 +41,32 @@ export default class DetailsOfDamagesPage {
   constructor(
     pageType: PageRequestType,
     private readonly placeOnReportService: PlaceOnReportService,
-    private readonly damagesSessionService: DamagesSessionService
+    private readonly damagesSessionService: DamagesSessionService,
+    private readonly reportedAdjudicationsService: ReportedAdjudicationsService
   ) {
     this.pageOptions = new PageOptions(pageType)
   }
 
   view = async (req: Request, res: Response): Promise<void> => {
     const { user } = res.locals
-    // draftId
     const adjudicationNumber = Number(req.params.adjudicationNumber)
     const damageToDelete = Number(req.query.delete) || null
-    const taskListUrl = adjudicationUrls.taskList.urls.start(adjudicationNumber)
-    const addDamagesUrl = adjudicationUrls.detailsOfDamages.urls.add(adjudicationNumber)
+    const isSubmittedEdit = this.pageOptions.displayAPIDataSubmitted() || this.pageOptions.displaySessionDataSubmitted()
+    const exitButtonHref = this.getExitUrl(req, adjudicationNumber, isSubmittedEdit)
 
-    const [draftAdjudicationResult, prisoner] = await Promise.all([
-      this.placeOnReportService.getDraftAdjudicationDetails(adjudicationNumber, user),
-      this.placeOnReportService.getPrisonerDetailsFromAdjNumber(adjudicationNumber, user),
-    ])
-    const { draftAdjudication } = draftAdjudicationResult
-    const reportedAdjudicationNumber = draftAdjudication.adjudicationNumber
-    const damages = this.getDamages(req, adjudicationNumber, draftAdjudication)
+    const { adjudication, prisoner } = await this.getAdjudicationAndPrisoner(adjudicationNumber, isSubmittedEdit, user)
+    const addDamagesUrl = isSubmittedEdit
+      ? `${adjudicationUrls.detailsOfDamages.urls.add(adjudicationNumber)}?submitted=true`
+      : `${adjudicationUrls.detailsOfDamages.urls.add(adjudicationNumber)}?submitted=false`
+
+    const redirectAfterRemoveUrl = isSubmittedEdit
+      ? `${adjudicationUrls.detailsOfDamages.urls.submittedEditModified(adjudicationNumber)}?delete=`
+      : `${adjudicationUrls.detailsOfDamages.urls.modified(adjudicationNumber)}?delete=`
+
+    const damages = this.getDamages(req, adjudicationNumber, adjudication)
 
     // If we are not displaying session data then fill in the session data
-    if (!this.pageOptions.displaySessionData()) {
+    if (this.pageOptions.displayAPIData() || this.pageOptions.displayAPIDataSubmitted()) {
```

Comment:
Spotted a couple of places in `detailsOfDamages`, `detailsOfEvidence`, `detailsOfWitnesses` which have  duplicated logic. Found these 3

```
this.pageOptions.displayAPIData() || this.pageOptions.displayAPIDataSubmitted()

const isSubmittedEdit = this.pageOptions.displaySessionDataSubmitted() || this.pageOptions.displayAPIDataSubmitted()

(this.pageOptions.displaySessionData() || this.pageOptions.displaySessionDataSubmitted())
```

Not a huge issue but I wonder if we could refactor these and centralise some of the logic? I thought perhaps could introduce an abstract class, probably worth having a team design session to chat about it?